react-native-ts/src/ItemListView.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {useCallback, useState, useEffect} from 'react';
+import {BSON} from 'realm';
+import {useUser} from '@realm/react';
+import {SafeAreaProvider} from 'react-native-safe-area-context';
+import {StyleSheet, Text, View} from 'react-native';
+import {Button, Overlay, ListItem} from 'react-native-elements';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import {CreateToDoPrompt} from './CreateToDoPrompt';
+import realmContext from './realmContext';
+import {Item} from './ItemSchema';
+const {useRealm, useQuery} = realmContext;
+
+Icon.loadFont(); // load FontAwesome font
+
+export function ItemListView() {
+  const realm = useRealm();
+  const items = useQuery(Item);
+  const user = useUser();
+  const [showNewItemOverlay, setShowNewItemOverlay] = useState(false);
+
+  useEffect(() => {
+    // initialize the subscriptions
+    const updateSubscriptions = async () => {
+      await realm.subscriptions.update(mutableSubs => {
+        if (user) {
+          // subscribe to all of the logged in user's to-do items
+          let ownItems = realm
+            .objects(Item)
+            .filtered(`owner_id == "${user.id}"`);
+          // use the same name as the initial subscription to update it
+          mutableSubs.add(ownItems, {name: 'ownItems'});
+        }
+      });
+    };
+    updateSubscriptions();
+  }, [realm, user]);
+
+  // createItem() takes in a summary and then creates an Item object with that summary
+  const createItem = useCallback(
+    ({summary}: {summary: string}) => {
+      // if the realm exists, create an Item
+      if (realm && user) {
```

Comment:
We're inconsistently checking the existence of realm. `useRealm()`'s signature assures us it will be a valid reference. If there's a case where it's not true, it should be updated in the SDK. If that's the case, maybe we should be a bit more strict than just silently ignoring the error state.

Reply:
We also know that we have to be logged in for this component to be mounted, so I'm also wondering about useUser() and the checks for user. This might be a case for `assert(user)` since mounting this view without a user might be a developer error?

Reply:
I've fixed the if statements.  I'll make an improvement to `@realm/react` later to fix the fact that user needs to be checked for `undefined` everywhere.  For now i'll just use the handy optional chaining syntax `user?.logout()`.